src/software/geom/util_test.cpp

```diff
@@ -1529,3 +1349,210 @@ TEST(GeomUtilTest, test_find_closest_point_many_points)
     Point reference_point(0.9, 0.9);
     EXPECT_EQ(test_points[1], findClosestPoint(reference_point, test_points));
 }
+
+TEST(GeomUtilTest, test_reduce_segments_collinear)
+{
+    Segment seg1 = Segment(Point(0, 1), Point(0, 2));
+    Segment seg2 = Segment(Point(0, 1.5), Point(0, 2.5));
+    Segment seg3 = Segment(Point(0, 3), Point(0, 4));
+    Segment seg4 = Segment(Point(0, 1.2), Point(0, 1.5));
+
+    Segment seg5 = Segment(Point(0, 6), Point(0, 9));
+    Segment seg6 = Segment(Point(0, 6), Point(0, 12));
+
+    Segment seg7 = Segment(Point(0, -2), Point(0, -5));
+
+    std::vector<Segment> segs = {seg1, seg2, seg3, seg4, seg5, seg6, seg7};
+
+    std::optional<std::vector<Segment>> reduced_segs =
+        combineToParallelSegments(segs, segs.front().toVector());
+
+    EXPECT_EQ(Segment(Point(0, 1), Point(0, 2.5)), reduced_segs.value()[0]);
+}
+
+TEST(GeomUtilTest, test_reduce_segments_perpendicular)
+{
+    Segment seg1 = Segment(Point(0, 0), Point(0, 10));
+    Segment seg2 = Segment(Point(0, 1), Point(1, 1));
+    Segment seg3 = Segment(Point(-5, -5), Point(5, -5));
+
+
+    std::vector<Segment> segs = {seg1, seg2, seg3};
+
+    std::optional<std::vector<Segment>> reduced_segs =
+        combineToParallelSegments(segs, segs.front().toVector());
+
+    EXPECT_EQ(reduced_segs->size(), 1);
+    EXPECT_EQ(reduced_segs->front().length(), 10);
+}
+
+
+TEST(GeomUtilTest, test_reduce_segments_perpendicular_and_parallel)
+{
+    Segment seg1 = Segment(Point(0, 0), Point(0, 10));
+    Segment seg2 = Segment(Point(0, 1), Point(1, 1));
+    Segment seg3 = Segment(Point(-5, -5), Point(5, -5));
+    Segment seg4 = Segment(Point(0, 2), Point(0, 15));
+    Segment seg5 = Segment(Point(0, 7), Point(9, 7));
+
+
+    std::vector<Segment> segs = {seg1, seg2, seg3, seg4, seg5};
+
+    std::optional<std::vector<Segment>> reduced_segs =
+        combineToParallelSegments(segs, seg1.toVector());
+
+    EXPECT_EQ(reduced_segs->size(), 1);
+    EXPECT_EQ(reduced_segs->front().length(), 15);
+    EXPECT_EQ(reduced_segs->front().getEnd().y(), 15);
+}
+
+TEST(GeomUtilTest, test_circle_tangent_rays)
+{
+    Point reference          = Point(0, 0);
+    Circle circle            = Circle(Point(0, 1), 0.5);
+    std::pair<Ray, Ray> test = getCircleTangentRaysWithReferenceOrigin(reference, circle);
+
+    EXPECT_EQ(test.second.getStart(), reference);
+    EXPECT_EQ(test.first.getStart(), reference);
+
+    EXPECT_NEAR(test.first.toUnitVector().x(), Vector(0.5, 0.866025).x(), 0.001);
```

Comment:
nit: Checking each ray is very close to another can be simplified with
`EXPECT_LT((test.first.toUnitVector() - Vector(your value)).len(), tolerance);`

Should be a bit easier to write / read than checking each component individually

Reply:
:+1: 
Changed